kernel/src/main/java/org/kframework/kprove/KProve.java

```diff
@@ -18,43 +18,28 @@
 import scala.Tuple2;
 
 import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
 import java.util.Set;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
 import static org.kframework.Collections.*;
 
-public class KProve {
-
-    public static final String BOUNDARY_CELL_PREFIX = "BOUND_";
-
+public record KProve(
+     KExceptionManager kem,
+     FileUtil files,
+     KPrint kprint,
+     KProveOptions kproveOptions,
+     CompiledDefinition compiledDefinition,
+     ProofDefinitionBuilder proofDefinitionBuilder,
+     Function<Definition, Rewriter> rewriterGenerator,
+     Stopwatch sw
+) {
     private static final int KPROVE_SUCCESS_EXIT_CODE = 0;
     private static final int KPROVE_MISMATCH_CONFIG_CODE = 1;
 
-    private final KExceptionManager kem;
-    private final FileUtil files;
-    private final KPrint kprint;
-    private final KProveOptions kproveOptions;
-    private final CompiledDefinition compiledDefinition;
-    private final BinaryLoader loader;
-    private final ProofDefinitionBuilder proofDefinitionBuilder;
-    private final Function<Definition, Rewriter> rewriterGenerator;
-    private final Stopwatch sw;
-
     @Inject
-    public KProve(KExceptionManager kem, FileUtil files, KPrint kprint, KProveOptions kproveOptions,
-                  CompiledDefinition compiledDefinition, BinaryLoader loader,
-                  ProofDefinitionBuilder proofDefinitionBuilder, Function<Definition, Rewriter> rewriterGenerator, Stopwatch sw) {
-        this.kem = kem;
-        this.files = files;
-        this.kprint = kprint;
-        this.kproveOptions = kproveOptions;
-        this.compiledDefinition = compiledDefinition;
-        this.loader = loader;
-        this.proofDefinitionBuilder = proofDefinitionBuilder;
-        this.rewriterGenerator = rewriterGenerator;
-        this.sw = sw;
-    }
+    public KProve {}
```

Comment:
Does this work?

Reply:
This is the new (slightly niche) syntax to apply an annotation to the implicitly-generated constructor of a record